change/@ni-nimble-components-c4841b4a-c42f-4d23-9748-857bbff400dd.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Support dynamic arbitrary header content",
```

Comment:
Jesse's framework question game me a separate thought that shouldn't impact this PR but makes me wonder if our sl-lib practice of wrapping components would be an issue.

Like lets say you have a shared icon column configuration where the shared wrapper sets up which icons to use for a systemlink app. The rendered dom in the page might look like:

```html
<nimble-table>
    <sl-icon-column record-field-name=""><!-- this might get emitted in the dom from the angular component wrapper? -->
        <nimble-column-icon record-field-name="">
             <!-- some configuration of a column that would be shared across multiple apps -->
             <nimble-icon-check slot="icon-1">
             <nimble-icon-other slot="icon-2">
        </nimble-column-icon>
    </sl-icon-column>
</nimble-table>
```

Reply:
Can't they just use `ng-template` for such a scenario?

Reply:
Oh, maybe? I know you can use ng-template as children in a component. But I'm not sure you can say my component itself behaves like an ng-template. ie the `sl-icon-column` above disappears at runtime and only it's children from the template stay.

But the same question could apply to blazor components wrappers if an app wants to make a re-usable configuration of an existing column type. (but again don't think it should necessarily impact anything going on in this PR atm)